src/frontend/src/common/queries/samples.ts

```diff
@@ -118,9 +119,13 @@ export async function createSamples({
       },
       sample: {
         [METADATA_KEYS_TO_API_KEYS.collectionDate]: collectionDate,
-        [METADATA_KEYS_TO_API_KEYS.collectionLocation]: collectionLocation,
+        [METADATA_KEYS_TO_API_KEYS.collectionLocation]: collectionLocation!.id,
```

Comment:
I don't recognize this notation. Does the `!` have special TS significance? Or is this meant to be a question mark?